lib/src/syn/v2/token/mod.rs

```diff
@@ -0,0 +1,385 @@
+//! Module specifying the token representation of the parser.
+
+use std::hash::Hash;
+
+mod keyword;
+pub(crate) use keyword::keyword_t;
+pub use keyword::Keyword;
+mod mac;
+pub(crate) use mac::t;
+
+use crate::sql::{language::Language, Algorithm};
+
+/// A location in the source passed to the lexer.
+#[derive(Clone, Copy, Eq, PartialEq, Hash, Debug)]
+pub struct Span {
+	/// Offset in bytes.
+	pub offset: u32,
+	/// The amount of bytes this location encompasses.
+	pub len: u32,
+}
+
+impl Span {
+	/// Create a new empty span.
+	pub const fn empty() -> Self {
+		Span {
+			offset: 0,
+			len: 0,
+		}
+	}
+
+	pub fn is_empty(&self) -> bool {
+		self.len == 0
+	}
+
+	/// Create a span that covers the range of both spans as well as possible space inbetween.
+	pub fn covers(self, other: Span) -> Span {
+		let start = self.offset.min(other.offset);
+		let end = (self.offset + self.len).max(other.offset + other.len);
+		let len = end - start;
+		Span {
+			offset: start,
+			len,
+		}
+	}
+}
+
+#[repr(u8)]
+#[derive(Clone, Copy, Eq, PartialEq, Hash, Debug)]
+pub enum Operator {
+	/// `!`
+	Not,
+	/// `+`
+	Add,
+	/// `-`
+	Subtract,
+	/// `÷`
+	Divide,
+	/// `×` or `∙`
+	Mult,
+	/// `||`
+	Or,
+	/// `&&`
+	And,
+	/// `<=`
+	LessEqual,
+	/// `>=`
+	GreaterEqual,
+	/// `*`
+	Star,
+	/// `**`
+	Power,
+	/// `=`
+	Equal,
+	/// `==`
+	Exact,
+	/// `!=`
+	NotEqual,
+	/// `*=`
+	AllEqual,
+	/// `?=`
+	AnyEqual,
+	/// `~`
+	Like,
+	/// `!~`
+	NotLike,
+	/// `*~`
+	AllLike,
+	/// `?~`
+	AnyLike,
+	/// `∋`
+	Contains,
+	/// `∌`
+	NotContains,
+	/// `⊇`
+	ContainsAll,
+	/// `⊃`
+	ContainsAny,
+	/// `⊅`
+	ContainsNone,
+	/// `∈`
+	Inside,
+	/// `∉`
+	NotInside,
+	/// `⊆`
+	AllInside,
+	/// `⊂`
+	AnyInside,
+	/// `⊄`
+	NoneInside,
+	/// `@123@`
+	Matches,
+	/// `+=`
+	Inc,
+	/// `-=`
+	Dec,
+	/// `+?=`
+	Ext,
+	/// `??`
```

Comment:
`Tco` and `Nco` are swapped . Also needs to be swapped in the `as_str` method.

(This is unrelated to the PR, but the `?:` operator is the same exact thing as the OR operator. Is there a point in having it in the language?)

Reply:
Good catch! I have fix the mistake.